tests/dir.props

```diff
@@ -77,6 +78,13 @@
     <RestorePackages>false</RestorePackages>
   </PropertyGroup>
 
+  <!-- If we want to overwrite the desired CoreCLR package version, we need to get the new version from the generated props file in bin/obj -->
+  <Import Condition="'$(OverwriteCoreClrPackageVersion)' == 'true'" Project="$(BaseIntermediateOutputPath)BuildVersion-$(OfficialBuildId).props" />
+
+  <PropertyGroup Condition="'$(OverwriteCoreClrPackageVersion)' == 'true'">
+    <PackageVersionArg>/p:CoreClrPackageVersion=2.1.0-preview1-$(BuildNumberMajor)-$(BuildNumberMinor)</PackageVersionArg>
```

Comment:
This will break when we stabilize or change the pre-release version information. I would suggest you try to get version from StableVersion property and if not available, get it from PackageVersion property. See https://github.com/dotnet/coreclr/blob/master/dir.props#L134 for details.